roles/manage_agents/tasks/detach_and_unlabel_all_removed_agents.yml

```diff
@@ -1,31 +1,29 @@
 ---
-- name: Number of agents being removed
-  ansible.builtin.set_fact:
-    manage_agents_removed_count: >
-      {{ manage_agents_allocated.resources | length - manage_agents_desired_count | int }}
-
-- name: Wait for all expected agents to be removed to be removed from the cluster
+- name: List agents currently labeled with the cluster order label
   kubernetes.core.k8s_info:
     kind: Agent
     api_version: agent-install.openshift.io/v1beta1
     namespace: "{{ default_agent_namespace }}"
     label_selectors:
       - "{{ cluster_order_label }}={{ manage_agents_cluster_order_name }}"
-      - "{{ esi_agent_resource_class_label }}={{ manage_agents_resource_class }}"
   register: manage_agents_allocated
-  until: >
-    (manage_agents_allocated.resources
-    | selectattr('spec.clusterDeploymentName.name', 'undefined')
-    | length) >= manage_agents_removed_count | int
-  retries: 10
-  delay: 30
 
 - name: List agents removed from the cluster
   ansible.builtin.set_fact:
     manage_agents_removed: >
       {{ manage_agents_allocated.resources | selectattr('spec.clusterDeploymentName.name', 'undefined') }}
 
-- name: Cleanup removed agents
+- name: Detach removed agents from cluster network
+  ansible.builtin.include_role:
+    name: massopencloud.esi.detach_network_from_node
+  vars:
+    node: "{{ item.metadata.annotations['esi.nerc.mghpcc.org/uuid'] }}"
+    network: "{{ manage_agents_cluster_network }}"
+  loop: "{{ manage_agents_removed }}"
+  loop_control:
+    label: "Detach agent {{ item.metadata.name }}"
+
```

Comment:
Just a note that we probably want to attach these agents back to the default `idle_agents` network, although it's unclear to me whether that network should be a default or not.